juriscraper/opinions/united_states/state/arizctapp_div_2.py

```diff
@@ -1,56 +1,48 @@
-# Scraper for Court of Appeals of Arizona, Division 2
-# CourtID: arizctapp_div_2
-# Court Short Name: arizctapp_div_2
-# Author: Andrei Chelaru
-# Reviewer: mlr
-# Date created: 23 July 2014
-
-
-import re
-from datetime import datetime
+"""
+Scraper for Court of Appeals of Arizona, Division 2
+CourtID: arizctapp_div_2
+Court Short Name: arizctapp_div_2
+Author: Andrei Chelaru
+Reviewer: mlr
+History:
+    2014-07-23: Created by Andrei Chelaru
+    2021-12-10: URL changed to recent opinions page
+"""
+
+from typing import List
 
 from juriscraper.lib.string_utils import clean_if_py3, titlecase
-from juriscraper.OpinionSite import OpinionSite
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
         # Feeling down and tired of of your regular life? Check out this website.
-        self.url = "https://www.appeals2.az.gov/ODSPlus/recentDecisions2.cfm"
-        self.base_path = "//*[@class='contentcontainer']//a[not(contains(., '(2)')) and not(contains(., 'page'))]"
-
-    def _get_case_names(self):
-        path = f"{self.base_path}/following::td[1]/*/text()"
-        return [titlecase(e) for e in self.html.xpath(path)]
-
-    def _get_download_urls(self):
-        path = f"{self.base_path}/@href"
-        return list(self.html.xpath(path))
-
-    def _get_case_dates(self):
-        path = f"{self.base_path}/following::td[2]/text()"
-        return list(map(self._return_case_date, self.html.xpath(path)))
-
-    @staticmethod
-    def _return_case_date(e):
-        e = re.sub(r"Opinion Filed:", "", clean_if_py3(e))
-        case_date = datetime.strptime(e.strip(), "%m/%d/%Y").date()
-        return case_date
-
-    def _get_precedential_statuses(self):
+        self.url = "https://www.appeals2.az.gov/ODSPlus/recentOpinionsHTML.cfm"
+        self.cases = []
+
+    def _process_html(self) -> None:
+        path = "//table//a[contains(@href, '.pdf')]"
+        for item in self.html.xpath(path):
+            docket = item.xpath("./text()")[0]
+            url = item.xpath("./@href")[0]
+            name = item.xpath("./following::td[1]/*/text()")[0]
+            # Expected: "Opinion Filed: 10/13/2021"
+            date_string = item.xpath("./following::td[2]/text()")[0].strip()
+            date = clean_if_py3(date_string).rsplit(":", 1)[1].strip()
+            summary_list = item.xpath("./following::tr[1]//text()")
+            summary = "".join(summary_list).strip()
+            self.cases.append(
+                {
+                    "name": titlecase(name),
+                    "date": date,
+                    "docket": docket,
+                    "url": url,
+                    "summary": summary,
```

Comment:
Should add status here.

Reply:
Done

Reply:
is this resolved?

Reply:
Yes, check the last commit @flooie

Reply:
It was committed at the time no?

Reply:
Tests for Python 3.7 - 3.10 failed due a library version conflict in the last commit. It's solved.
It's probably why you couldn't fetch the last commit @flooie 
Please, try again.

Reply:
Just make it clear, there are **2 methods to add the precential status** to opinions scraped with super class `OpinionSiteLinear`: adding the status per opinion or per singular global status (`self.status`).

https://github.com/freelawproject/juriscraper/blob/e6a820177d2015a9578d7f601b3054d7cede3822/juriscraper/OpinionSiteLinear.py#L31-L46

Reply:
this is an interesting way to do it but lets not assign the status that way.  Let's put it in with the rest of the data.

Reply:
Done @flooie